### PR TITLE
update cli documentation

### PR DESCRIPTION
**Pull request by fbourigault**
This update the CLI documentation which looked out of sync with the `composer --help output`.

Thanks to proofreaders :)

### CODE REVIEW

File: doc/03-cli.md [hunk 1]
```diff
@@ -328,18 +333,19 @@ php composer.phar show monolog/monolog 1.0.2
...
 * **--self (-s):** List the root package info.
-* **--tree (-t):** List your dependencies as a tree. If you pass a package name it will show the dependency tree for that package.
 * **--name-only (-N):** List package names only.
 * **--path (-P):** List package paths.
+* **--tree (-t):** List your dependencies as a tree. If you pass a package name it will show the dependency tree for that package.
+* **--latest (-l):** List all installed packages including their latest version.
 * **--outdated (-o):** Implies --latest, but this lists *only* packages that have a newer version available.
 * **--minor-only (-m):** Use with --latest. Only shows packages that have minor SemVer-compatible updates.
 * **--direct (-D):** Restricts the list of packages to your direct dependencies.
-* **--format (-f):** Lets you pick between text (default) or json output format.
```
Comment (alcohol) [inline doc/03-cli.md]: Any particular reason you dropped the `--format` option of the `show` and `outdated` commands?

Comment (fbourigault) [inline doc/03-cli.md]: I made a mistake on this one!

Comment (fbourigault) [inline doc/03-cli.md]: Fixed :)

File: doc/03-cli.md [hunk 2]
```diff
@@ -135,30 +137,31 @@ php composer.phar update vendor/*
...
   terminals or scripts which don't handle backspace characters.
 * **--no-suggest:** Skips suggested packages in the output.
+* **--with-dependencies:** Add also all dependencies of whitelisted packages to the whitelist.
 * **--optimize-autoloader (-o):** Convert PSR-0/4 autoloading to classmap to get a faster
   autoloader. This is recommended especially for production, but can take
   a bit of time to run so it is currently not done by default.
 * **--classmap-authoritative (-a):** Autoload classes from the classmap only.
   Implicitly enables `--optimize-autoloader`.
 * **--apcu-autoloader:** Use APCu to cache found/not-found classes.
-* **--lock:** Only updates the lock file hash to suppress warning about the
```
Comment (stof) [inline doc/03-cli.md]: why are you moving the doc around ?

Comment (fbourigault) [inline doc/03-cli.md]: I sorted options to get the same order as displayed by the help output.

**Comment by Seldaek** [comment]
Thanks!

**Pull request merged**